dor/providers/op_client.py

```diff
@@ -0,0 +1,21 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+
+@dataclass
+class FileSetSearchResult:
+    file_set_identifier: str
+    bin_identifier: str
+
+
+class OPClient(ABC):
+
+    @abstractmethod
+    def search_for_file_set(self, file_set_identifier: str) -> FileSetSearchResult | None:
```

Comment:
Eventually I think we'll need to submit the `bin_identifier` and maybe `alternate_identifier` as well for the full lookup, but this is fine for now.